operator-common/src/main/java/io/strimzi/operator/common/Util.java

```diff
@@ -128,6 +130,31 @@ public static RuntimeException missingSecretException(String namespace, String s
         return new RuntimeException("Secret " + namespace + "/" + secretName + " does not exist");
     }
 
+    /**
+     * Returns exception when secret data is missing a particular item. This is used from several different methods to provide identical exception.
+     *
+     * @param namespace     Namespace of the Secret
+     * @param secretName    Name of the Secret
+     * @param dataFieldName Name of the Secret field
+     * @return              RuntimeException
+     */
+    public static RuntimeException missingDataInSecretException(String namespace, String secretName, String dataFieldName) {
+        return new RuntimeException("The Secret " + namespace + "/" + secretName + " is missing the field " + dataFieldName);
+    }
+
+    /**
+     * Returns exception when certificate is corrupt. This is used from several different methods to provide identical exception.
+     *
+     * @param namespace     Namespace of the Secret
+     * @param secretName    Name of the Secret
+     * @param keyName       Name of the Secret key
+     * @return              RuntimeException
+     */
+    public static RuntimeException corruptCertificateException(String namespace, String secretName, String keyName) {
+        return new RuntimeException("Bad/corrupt certificate found in data." + keyName + ".crt of Secret "
+                + secretName + " in namespace " + namespace);
+    }
```

Comment:
These methods seem strange. They do not seem to contain any real logic and seem to be used from a few places only - the places where they are used also do not seem to use the _central facilities_ for whatever reason. Would it be better to just inline this or create a new exception types and use this in their constructors?

Reply:
I wrote these methods here based on the method `missingSecretException` that was already in this class. I can create a new exception type if that is the preferred way to handle these kinds of exceptions.

Reply:
I've moved these inline